source/gui/servergui.cpp

```diff
@@ -50,696 +50,753 @@
-    UTFString msgf;
+typedef struct log_queue_element_t
+{
+   int       level;
+   UTFString msg;
+   UTFString msgf;
 } log_queue_element_t;
 
 // Define a new frame type: this is going to be our main frame
 class MyDialog : public wxDialog {
 public:
-    // ctor(s)
-    MyDialog(const wxString &title, MyApp *_app);
+// ctor(s)
+MyDialog(const wxString &title, MyApp *_app);
 
-    void logCallback(int level, UTFString msg, UTFString msgf); //called by callback function
+void logCallback(int level, UTFString msg, UTFString msgf);     //called by callback function
 private:
-    MyApp *app;
-    wxButton *startBtn, *stopBtn, *exitBtn, *pubipBtn;
-    wxTextCtrl *txtConsole, *ipaddr, *port, *slots, *passwd, *terrain, *sname, *logfilename, *adminuid;
-    wxComboBox *smode, *logmode;
-    wxNotebook *nbook;
-    wxListCtrl *slotlist;
-    wxScrolledWindow *settingsPanel;
-    wxPanel *logPanel, *playersPanel;
-    wxTimer *timer1;
-    wxToolBar *tb;
+MyApp            *app;
+wxButton         *startBtn, *stopBtn, *exitBtn, *pubipBtn;
+wxTextCtrl       *txtConsole, *ipaddr, *port, *slots, *passwd, *terrain, *sname, *logfilename, *adminuid;
+wxComboBox       *smode, *logmode;
+wxNotebook       *nbook;
+wxListCtrl       *slotlist;
+wxScrolledWindow *settingsPanel;
+wxPanel          *logPanel, *playersPanel;
+wxTimer          *timer1;
+wxToolBar        *tb;
 
-    void OnQuit(wxCloseEvent &event);
+void OnQuit(wxCloseEvent &event);
 
-    int loglevel;
-    int secondsCounter;
-    pthread_t notifyThread;
-    bool server_is_running;
+int                             loglevel;
+int                             secondsCounter;
+pthread_t                       notifyThread;
+bool                            server_is_running;
 
-    Mutex log_queue_mutex;
-    std::deque<log_queue_element_t> log_queue;
+Mutex                           log_queue_mutex;
+std::deque<log_queue_element_t> log_queue;
 
-    void OnBtnStart(wxCommandEvent &event);
+void OnBtnStart(wxCommandEvent &event);
 
-    void OnBtnStop(wxCommandEvent &event);
+void OnBtnStop(wxCommandEvent &event);
 
-    void OnBtnExit(wxCommandEvent &event);
+void OnBtnExit(wxCommandEvent &event);
 
-    void OnBtnPubIP(wxCommandEvent &event);
+void OnBtnPubIP(wxCommandEvent &event);
 
-    void OnContextMenu(wxContextMenuEvent &event);
+void OnContextMenu(wxContextMenuEvent &event);
 
-    int startServer();
+int startServer();
 
-    int stopServer();
+int stopServer();
 
-    void updateLogTab();
+void updateLogTab();
 
-    void updatePlayerList();
+void updatePlayerList();
 
-    void OnTimer(wxTimerEvent &event);
+void OnTimer(wxTimerEvent &event);
 
-    DECLARE_EVENT_TABLE()
+DECLARE_EVENT_TABLE()
 };
 
-enum {
-    btn_start, btn_stop, btn_exit, EVT_timer1, btn_pubip, ctxtmenu_kick, ctxtmenu_ban, ctxtmenu_pm
+enum
+{
+   btn_start, btn_stop, btn_exit, EVT_timer1, btn_pubip, ctxtmenu_kick, ctxtmenu_ban, ctxtmenu_pm
 };
 
 BEGIN_EVENT_TABLE(MyDialog, wxDialog
-)
+                  )

```

Comment:
Please hand-edit this and put the closing brackets on the same line as the macro :smile: